### DISCUSSION TITLE
Learning Java — Error please help. I/O error.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#141067] [Programming Help] [answered]
### Body

```java
import java.util.Scanner;
public class Main{
    public static void main(String[] args){
        System.out.println("Enter your name > ");
        Scanner username = new Scanner(System.in);
        System.out.println(username);
    }
}
 ```
 

Enter your name > 
java.util.Scanner[delimiters=\p{javaWhitespace}+][position=0][match valid=false][need input=false][source closed=false][skipped=false][group separator=\,][decimal separator=\.][positive prefix=][negative prefix=\Q-\E][positive suffix=][negative suffix=][NaN string=\QNaN\E][infinity string=\Q∞\E]

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by d-coder111** [comment, marked as answer]
Hey @WorldsDumbestCoder, Here's the corrected code:


import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        System.out.print("Enter your name > ");
        Scanner username = new Scanner(System.in);
        String name = username.nextLine();  // Get the input as a string
        System.out.println("Your name is: " + name);
    }
}


Feel free to mark it as the accepted answer if you found it helpful!

**Comment by Arx-F** [comment]
The issue here is that you're printing the `Scanner` object itself, which is why you see the complex output that represents the internal state of the `Scanner`. Instead, you should be printing the user's input, not the `Scanner` object.

To fix this, you can use `username.nextLine()` to read the actual input provided by the user. Here's the corrected version of your code:

```java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        System.out.println("Enter your name > ");
        Scanner username = new Scanner(System.in);
        String name = username.nextLine();  // Get the user's input
        System.out.println("Hello, " + name + "!");  // Print the input
    }
}
```

### Explanation:
- `username.nextLine()` reads the entire input entered by the user.
- Instead of printing the `Scanner` object, we store the user's input in a `String` variable (`name`) and print that.

This will give you the expected result where the program asks for your name and then prints a greeting with it.

**Reply by ghost** [reply]
I already gave up on coding and I am going to delete my GitHub account after my class.

**Reply by Arx-F** [reply]
Why bro nooo :(

**Reply by ghost** [reply]
```java
public class Main{
  public static void main (String[] args){
     System.out.println(“I heard you need a high amount of comprehension that far surpasses mine, so should I just give up?”);
 }
}
```

I know this will output “I heard you need a high amount of comprehension that far surpasses mine.

**Reply by KBluelvl** [reply]
Actually, that’s not the correct output. The output looks more like this:
"I heard you need a high amount of comprehension that far surpasses mine, so should I just give up?".
But don't give up man !

**Reply by ghost** [reply]
[Touche](https://www.youtube.com/watch?v=2ZxlGaPvQQA) I can also do some markdown
